docs/modules/ROOT/pages/property-source-config/configmap-propertysource.adoc

```diff
@@ -193,8 +193,26 @@ spring:
 ----
 
 This will search for every configmap in namespace `spring-k8s` that has labels `{letter : a}`. The important
-thing to notice here is that unlike reading a configmap by name, this can result in _multiple_ config maps read.
-As usual, the same feature is supported for secrets.
+thing to notice here is that unlike reading a configmap by name, this can result in _multiple_ config maps read. The same feature is supported for secrets.
+
+Another important point is that if you plan to use `labels` searching, we strongly encourage such configmaps (and secrets) names to have zero or one dashes (`-`) (as defined in the `.metadata.name`). The reason for this is simple, suppose you define two configmaps that have the same labels:
```

Comment:
But we know what the active profiles are right?  For in you example, if that active profile is 'dev' and the app name is `my-app` then we return both.  If the active profile is `dev` and the app name is `my` then we don't return either.  If the active profile is `app-dev` and the app name is `my` we return only `my-app-dev`.

Reply:
OK now I see.  Do we try and use the profile and or application name today when using secrets?

Reply:
So when you say we don't care about the name or profile when using labels is that currently or what you are proposing we change in this PR?

Reply:
Yeah or this could do something like this
```
        sources:
          - labels:
              letter: a
              profile: ${spring.profiles.active}
```

At least I would imagine that would work, might need to test that.

Reply:
We will have to maintain the current behavior so to break it, it will have to wait for a major.